feat: enables setting zarf vars with env vars (#264)
defenseunicorns/uds-cli: README.md

In the example above, the `OUTPUT` variable is created as part of a Zarf Action in the [output-var](src/test/packages/zarf/no-cluster/output-var) package, and the [receive-var](src/test/packages/zarf/no-cluster/receive-var) package expects a variable called `OUTPUT`.

### Variable Precedence and Specificity
In a bundle, variables can come from 4 sources. Those sources and their precedence are shown below in order of least to most specificity:
- Variables declared in a Zarf pkg
- Variables `import`'ed from a bundle package's `export`
- Variables configured in `uds-config.yaml`
- Variables set with an environment variable prefixed with `UDS_` (ex. `UDS_OUTPUT`)

> That is to say, deploy-time variables declared in `uds-config.yaml` take precedence over all other variable sources.

Reviewer: would env variables with `UDS_` prefix now take precedence?

Author: For sure! The list goes from least specific to most specific, so `UDS_` env vars take precedence over anything else

In the example above, the `OUTPUT` variable is created as part of a Zarf Action in the [output-var](src/test/packages/zarf/no-cluster/output-var) package, and the [receive-var](src/test/packages/zarf/no-cluster/receive-var) package expects a variable called `OUTPUT`.

### Variable Precedence and Specificity
In a bundle, variables can come from 4 sources. Those sources and their precedence are shown below in order of least to most specificity:
- Variables declared in a Zarf pkg
- Variables `import`'ed from a bundle package's `export`
- Variables configured in `uds-config.yaml`
- Variables set with an environment variable prefixed with `UDS_` (ex. `UDS_OUTPUT`)

> That is to say, deploy-time variables declared in `uds-config.yaml` take precedence over all other variable sources.

Reviewer: ```suggestion
That is to say, Variables set with an environment variable prefixed with `UDS_` take precedence over all other variable sources.
```
